### PR TITLE
Doc: use built_in_by_default and build_dependencies directive

### PR DESCRIPTION
**Pull request by elil**
## What does this PR do?
Start populating new table at https://gdal.org/drivers/raster/index.html

## Tasklist

 - [ ] More to go on the table just seeing if this is the right track
 - [ ] No test case(s)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Notes

I didn't build the sphinx docs so I'm not certain this is a good commit but it probably is.

### CODE REVIEW

**Comment by mloskot** [comment]
Thanks for this overhaul.
The docs build fine, https://dev.azure.com/osgeo/gdal/_build/results?buildId=1416, but there is no way to CI-check the presentation.

FYI, building docs is pretty much trivial sequence of commands on Linux (on Windows it's too terrible either):

https://github.com/OSGeo/gdal/blob/62f62580226ca28838801174b6c130ca93ae4294/gdal/doc/.azure-pipelines.yml#L27-L40

I will try to build the docs locally and check this PR as soon as I have access to Linux.

File: gdal/doc/source/drivers/raster/bsb.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ BSB -- Maptech/NOAA BSB Nautical Chart Format
 
 .. shortname:: BSB
 
+.. built_in_by_default:: (can be disabled)
```
Comment (rouault) [inline gdal/doc/source/drivers/raster/bsb.rst]:
```suggestion
.. built_in_by_default::
```
I suggest to remove the " (can be disable)". This is now true for all formats, at least with Unix builds

File: gdal/doc/source/drivers/raster/grib.rst [hunk 2]
```diff
@@ -6,6 +6,8 @@ GRIB -- WMO General Regularly-distributed Information in Binary form
 
 .. shortname:: GRIB
 
+.. built_in_by_default:: (can be disabled)
```
Comment (rouault) [inline gdal/doc/source/drivers/raster/grib.rst]:
```suggestion
.. built_in_by_default::
```

File: gdal/doc/source/drivers/raster/jp2openjpeg.rst [hunk 3]
```diff
@@ -8,6 +8,8 @@ JP2OpenJPEG -- JPEG2000 driver based on OpenJPEG library
 
 .. build_dependencies:: openjpeg >= 2.1
 
+.. build_dependencies:: OpenJPEG library (v2)
```
Comment (rouault) [inline gdal/doc/source/drivers/raster/jp2openjpeg.rst]:
```suggestion
```
To remove: already indicated before

File: gdal/doc/source/drivers/raster/mbtiles.rst [hunk 4]
```diff
@@ -6,6 +6,8 @@ MBTiles
 
 .. shortname:: MBTiles
 
+.. build_dependencies:: OGR SQLite driver
```
Comment (rouault) [inline gdal/doc/source/drivers/raster/mbtiles.rst]:
```suggestion
.. build_dependencies:: libsqlite3```

**Comment by mloskot** [comment]
AFAICT, this works and looks well

image

File: gdal/doc/source/drivers/raster/mbtiles.rst (hunk 4 above)
Comment (elil) [reply gdal/doc/source/drivers/raster/mbtiles.rst]: I assumed that ``` was a typo so my current PR reverses this.  If it wasn't a typo, you'll need to correct it again.

Comment (rouault) [reply gdal/doc/source/drivers/raster/mbtiles.rst]: ah, yes ``` is a markdown quoting thing. Not idented to end up in rst

**Pull request merged**